src/documents/TextfieldAPI.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Textfield
+layout: default
+category: Components
+---
+
+The **Textfield** is small component for adding aria attributes and data binding to a HTML textfield. Typically this will be used in conjunction with a controller (e.g. [**Textfield Slider**](TextfieldSlider.md), [**Textfield Stepper**](TextfieldStepper.md)) to restrict the values and pair with additional input methods.
+
+## Creator ##
+
+Use the following function to create a Textfield component:
+
+<table>
+    <tbody>
+        <tr>
+            <th>Method</th>
+            <td>
+                <code>fluid.textfield(container, options);</code>
+            </td>
+        </tr>
+        <tr>
+            <th>Description</th>
+            <td>
+                Instantiates the textfield component. Provides databinding between the HTML `<input>` and the components model. Also provides options for setting an `"aria-label"` and/or `"aria-labelledby"` attribute.
+            </td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
+            <td>
+                <dl>
+                    <dt>container</dt>
+                    <dd>
+                        A CSS-based selectors, single-element jQuery object, or DOM element that identifies HTML `<input>` to bind the Textfield component to.
+                    </dd>
+                </dl>
+                <dl>
+                    <dt>options</dt>
+                    <dd>
+                        An optional data structure that configures the Textfield component, as described below.
+                    </dd>
+                </dl>
+            </td>
+        </tr>
+        <tr>
+            <th>Returns</th>
+            <td>The Textfield component</td>
+        </tr>
+        <tr>
+            <th>Examples</th>
+            <td>
+<pre>
+<code>
+var textfield = fluid.textfield(".flc-textfield", {
+    model: {
+        value: "Hello World"
+    },
+    strings: {
+        "label": "Insert Demo Name"
+    }
+});
+</code>
+</pre>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Methods ##
+
+### setModel ###
+
+<table>
+    <tbody>
+        <tr>
+            <th>Method</th>
+            <td>
+                <code>textfield.setModel(event);</code>
+            </td>
+        </tr>
+        <tr>
+            <th>Description</th>
+            <td>
+                The <code>setModel</code> method allows for updating the model value based on an event. This is primarily used internally for binding to the input's change event.
+            </td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
+            <td>
+                <dl>
+                    <dt>event</dt>
+                    <dd>
+                        An event object including a change value.
```

Comment:
Would be helpful to provide more info about what this event is (a jQuery event with a link to jquery event system documentation) and from where the function fetches data from (`event.target.value`).